### PR TITLE
feat(experimental): Parse match expressions

### PR DESCRIPTION
**Pull request by jfecher**
# Description

## Problem\*

Working towards #988 

## Summary\*

Add parser, formatter, lsp support for match expressions

## Additional Context

Still can't use these in a program, we panic in the monomorphizer for them. Just pushing these somewhat trivial changes before the chunkier changes of compiling match expressions to a decision tree, checking for completeness, redundancy, then monomorphizing etc.

## Documentation\*

Check one:
- [ ] No documentation needed.
- [ ] Documentation included in this PR.
- [x] **[For Experimental Features]** Documentation to be submitted in a separate PR.

# PR Checklist\*

- [x] I have tested the changes locally.
- [x] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

**Comment by jfecher** [comment]
(ready for review now)

File: tooling/nargo_fmt/src/formatter/expression.rs [hunk 1]
```diff
@@ -895,6 +898,68 @@ impl<'a, 'b> ChunkFormatter<'a, 'b> {
...
+        group.tag = Some(group_tag);
+        group.force_multiple_lines = true;
+
+        group.text(self.chunk(|formatter| {
+            formatter.write_keyword(Keyword::Match);
+            formatter.write_space();
+        }));
+
+        self.format_expression(match_expression.expression, &mut group);
+        self.skip_comments_and_whitespace_chunk();
```
Comment (asterite) [inline tooling/nargo_fmt/src/formatter/expression.rs:926]:
```suggestion
        group.trailing_comment(self.skip_comments_and_whitespace_chunk());
```

Comment (asterite) [reply tooling/nargo_fmt/src/formatter/expression.rs:926]: Otherwise the chunk is lost... I know there are no tests for this but this would be the correct way, and in general the tests for comments (after the initial formatter PR) are added as we find bugs for edge cases, so no tests for now is okay 😊

**Review by asterite — APPROVED** [review]
Looks good!
